README


This archive contains the distribution Padre-Plugin-Perl6,
version 0.70:

  Perl 6 Support for Padre

This software is copyright (c) 2010 by Ahmad M. Zawawi.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Changes

0.70   2010.12.06
	Restored missing POT/PO translation sources (Damyan Ivanov, AZAWAWI)
	Renamed method 'show_preferences' to 'plugin_preferences' (ZENOG)
	Use App::Grok when you actually need it in Perl 6 Help (AZAWAWI)

0.69   2010.10.31
	Fixed a regression in the highlighter and syntax checker (AZAWAWI)

0.68   2010.10.29
	Removed Test::NeedsDisplay dependency (AZAWAWI)

0.67   2010.10.27
	t/00-load.t is magically injected by [@PadrePlugin] via [LoadTests] (AZAWAWI)

0.66   2010.10.23
	Switch to Dist::Zilla
	Removed "Update Six" experimental feature
	Removed "Perl refactor" undeveloped feature
	Fixed a syntax error in "Create Perl 6 Module" template
	Fixed crashes while exporting document to HTML

0.65 2010.10.19

	No need for cleanup STD lex cache anymore
	Translation files are now installed via Module::Build 0.36 share_dir
	Fixed a Padre compatibility problem with the syntax checker and outline
	Renamed some packages to more Perl-ish names
	Outline is working again with the latest STD.pm for package declarators

0.64 2010.03.24

	Depends on Padre 0.57 for various module naming changes
	Fixed ticket #883: Opening an existing Perl 6 script Hangs Padre on win32

0.63 2010.02.24

	Depends on Padre 0.56 and later
	Depends on S:H:P6 0.79 for the latest STD innovation
	Fixed Help provider API change that was introduced in Padre 0.56
	Ignore MYMETA.yml
	Fixed the out-of-memory Storable nfreeze/thaw bug by using YAML::XS instead

0.62 2010.01.07

	Depends on S:H:P6 v0.77 for the latest STD
	Replace Padre::Debug by Padre::Logger

0.61 2009.12.10

	Replaced debug called with new TRACE() call

0.60 2009.9.18

	Depends now on Net::HTTP, HTTP::Status and Archive::Zip
	Depends on S:H:P6 v0.74 for the latest STD
	First attempt at upgrading Six distribution on win32 via the web
	"Update Six" is now in the maintenance menu above "Cleanup std lex cache"
	When "update six" is selected, a dialog is shown and when a release is selected,
		it is downloaded & then unzipped in c:\strawberry. This is at
		the moment a win32-only functionality.
	The Six updater is a background task that downloads six distros, backups
		up old six and unzips the new downloaded six and it outputs its
		progress meanwhile to the output panel
	The Six updater dialog features a new cool camelia-enabled look and a 
	"View Release Notes" button

0.59 2009.9.8

	Find perl6.exe in PDX and later Six releases and stay compatible with 0.41
	print -> Padre::Util::debug

0.58 2009.8.24

	Updated translations:
		Arabic
		French (jquelin++)
		German (Sewi++)
	Integrated with Padre's 0.44 Quick Fix Helper
	Integrated with Padre's 0.44 Help Provider
	Reverting back to _sharedir until we get the Module::Build gets a sharedir property
	STD.pm error prefix was recently switched from ### PARSE FAILED ### to ===SORRY!===
	Prevent STD from inserting that block unicode character into error messages. $PERL6HERE
	Removed "Perl 6 Executable" until rakudo actually supports it again
	Depends on Syntax:Highlight:Perl6 0.72
	Depends on Perl6::Doc 0.45
	Added perlobjintro and perlreintro that Perl6::Doc 0.45 now supports

0.57 2009.8.17

	Depends on Padre 0.43 to make use of plugin_directory_share
	Depends on S:H:P6 0.71 and App::Grok 0.19
	Perl 6 Help dialog is moved to Padre core as "Help Search"
	Create Perl 6 menu is working again. (szabgab)
	Reuse Padre::Wx::Main help method (szabgab)
	Updated translations:
		German translation update (Sewi++)
		French translation (jquelin++)
		dutch translation (ddn123456++)
	Renamed to the new Padre 0.43 PO naming scheme
	use the new Padre::Perl API for finding perl5 (szabgab)
	safety crash hiding (szabgab)
	double quote temporary directory

0.56 2009.7.28

	- Updated Translations:
		Arabic
		Brazilian portuguese (garu++)
		Japanese (ishigaki++)
	- Camelia icon on all dialogs.
	- use Padre::Constant::WIN32 in all locations.
	- IRC Live support menu items are simpler.
	- Added an easy-to-use menu option to access perlintro and perlsyn directly (Hinrik++).
	- Depends on App::Grok 0.17 for more bug fixes/enhancements..

0.55 2009.7.25

	- This is a bugfix release for Padre's Birthday Six Release.
	- Removed commented highlighter, and tidied the files using Padre's
		perltidyrc.
	- Fixed missing use (szabgab++).
	- Grok target location is now shown in the dialog's title.
	- Fixed the environment detection code, disable Perl 6 Executable menu
		entry if there's no parrot.exe (adamk++).
	- Depends on App::Grok 0.14 for less dependencies.
	- Find the path to hard perl6.exe hard coded in strawberry six (szabgab++).
	- return the path to the perl6 executable if it exists (szabgab++).

0.54 2009.7.24

	- Depends on S:H:P6 version 0.65 to fix ticket:461 "Syntax check hanging
		with perl6 infix & regex"
	- Depends on Padre 0.41. Please upgrade it or otherwise it will not work.
	- Allow the use of the new syntax highlighter configuration of Padre 0.41 (szabgab++)
	- Move the Parrot based syntax highlighting to the Parrot plugin (szabgab++)
	- F2 now works on select text - if there is any (szabgab++)
	- Updated translations:
		- Dutch (ddn123456++)
		- French (jquelin++)
	- move the text_with_on_nl method from the Perl 6 document to the Padre::Document
	- Remove obsolete colorizer selection list. It is now handled by Padre.
	- Updated dutch translation.

0.53 2009.7.18

	- Moved Rakudo Perl 6 functionality into a separate menu.
	- Fixed ticket:457 "Store Perl 6 temporary folders in a persistent
		temporary directory"
	- Cleanup STD lex cache now works with ticket:457
	- Updated Arabic translation.

0.52 2009.7.18

	- Removed parse_perl6_table_index.pl & perl6_table_index.pod since this is
		grok's territory now...
	- Fixed a couple of Perl6->Perl 6 typos, and added some missing POD.
	- Unified copyright on all files.
	- Added 03-copyright.t to author-only test for copyright.
	- Moved Cleanup STD Lex Cache into "Maintenance" sub menu/
	- #padre, #perl6 and perl6-projects.org are now accessible via the "More Help?" menu

Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'Locale::Msgfmt' => '0.15',
    'Module::Build' => '0.3601',
    'Test::More' => '0'
  },
  'configure_requires' => {
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'Perl 6 Support for Padre',
  'dist_author' => [
    'Ahmad M. Zawawi <ahmad.zawawi@gmail.com>',
    'Gabor Szabo L<http://szabgab.com/>'
  ],
  'dist_name' => 'Padre-Plugin-Perl6',
  'dist_version' => '0.70',
  'license' => 'perl',
  'module_name' => 'Padre::Plugin::Perl6',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'App::Grok' => '0.22',
    'Class::XSAccessor' => '1.02',
    'File::Which' => '0.05',
    'Padre' => '0.72',
    'Perl6::Doc' => '0.45',
    'Syntax::Highlight::Perl6' => '0.87',
    'URI' => '1.37',
    'perl' => '5.010'
  },
  'script_files' => [],
  'share_dir' => {
    'dist' => 'share'
  }
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
